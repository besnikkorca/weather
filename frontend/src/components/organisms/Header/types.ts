export interface Props {
  active: string;
  setActive: (active: string) => void;
}
