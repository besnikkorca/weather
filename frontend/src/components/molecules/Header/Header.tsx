import { Component, ReactNode } from 'react';
import List from 'src/components/atoms/List';
import ListItem from 'src/components/atoms/ListItem';
import styles from './Header.module.scss';
import { Props } from './types';

export default class Header extends Component<Props> {
  render(): ReactNode {
    return (
      <header className={styles.header}>
        <nav>
          <List>
            {['Ottawa', 'Moscow', 'Tokyo'].map((city) => (
              <ListItem
                key={city}
                active={city === this.props.active}
                onClick={this.props.setActive}
              >
                {city}
              </ListItem>
            ))}
          </List>
        </nav>
      </header>
    );
  }
}
