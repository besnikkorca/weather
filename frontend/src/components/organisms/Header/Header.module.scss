.header {
  padding: 0rem 1.5rem 3rem;
}
