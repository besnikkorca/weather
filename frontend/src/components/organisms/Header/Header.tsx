import { Component, ReactNode } from 'react';
import CitiesList from 'molecules/CitiesList';
import styles from './Header.module.scss';
import { Props } from './types';

export default class Header extends Component<Props> {
  render(): ReactNode {
    const { cities, active, setActive } = this.props;
    return (
      <header className={styles.header}>
        <nav>
          <CitiesList active={active} setActive={setActive} cities={cities} />
        </nav>
      </header>
    );
  }
}
