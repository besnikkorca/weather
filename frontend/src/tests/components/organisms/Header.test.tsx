import { screen, render } from '@testing-library/react';
import Header from 'src/components/organisms/Header';
import cities from '../../toydata/citiesList';

describe('Header', () => {
  it('should render', () => {
    const { container } = render(<Header cities={cities} active="New York" setActive={() => {}} />);
    expect(container).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const view = render(<Header cities={cities} active="Ottawa" setActive={() => {}} />);

    expect(view).toMatchSnapshot();
  });

  it('renders with Ottawa active', () => {
    render(<Header cities={cities} active="Ottawa" setActive={() => {}} />);

    expect(screen.getByText('Ottawa')).toHaveClass('active');
  });

  it('renders with Moscow active', () => {
    render(<Header cities={cities} active="Moscow" setActive={() => {}} />);

    expect(screen.getByText('Moscow')).toHaveClass('active');
  });

  it('renders with Tokyo active', () => {
    render(<Header cities={cities} active="Tokyo" setActive={() => {}} />);

    expect(screen.getByText('Tokyo')).toHaveClass('active');
  });

  it('should call setActive when clicked', () => {
    const mockSetActive = jest.fn();
    render(<Header cities={cities} active="Ottawa" setActive={mockSetActive} />);

    screen.getByText('Moscow').click();

    expect(mockSetActive).toBeCalledTimes(1);
  });
});
