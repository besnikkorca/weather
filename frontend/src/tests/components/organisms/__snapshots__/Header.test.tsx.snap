// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <nav
          class="nav"
        >
          <ul
            class="list"
          >
            <li
              class="listItem active"
            >
              Ottawa
            </li>
            <li
              class="listItem notActive"
            >
              Moscow
            </li>
            <li
              class="listItem notActive"
            >
              Tokyo
            </li>
            <li
              class="listItem notActive"
            >
              London
            </li>
            <li
              class="listItem notActive"
            >
              New York
            </li>
          </ul>
          <div
            class="toggle"
          >
            <label
              class="switch"
            >
              <input
                type="checkbox"
              />
              <span
                class="slider round"
              />
            </label>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <nav
        class="nav"
      >
        <ul
          class="list"
        >
          <li
            class="listItem active"
          >
            Ottawa
          </li>
          <li
            class="listItem notActive"
          >
            Moscow
          </li>
          <li
            class="listItem notActive"
          >
            Tokyo
          </li>
          <li
            class="listItem notActive"
          >
            London
          </li>
          <li
            class="listItem notActive"
          >
            New York
          </li>
        </ul>
        <div
          class="toggle"
        >
          <label
            class="switch"
          >
            <input
              type="checkbox"
            />
            <span
              class="slider round"
            />
          </label>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
